chronos/transports/common.nim

```diff
@@ -625,3 +631,72 @@ when defined(windows):
        {.stdcall, dynlib: "kernel32", importc: "SetNamedPipeHandleState".}
   proc resetEvent*(hEvent: HANDLE): WINBOOL
        {.stdcall, dynlib: "kernel32", importc: "ResetEvent".}
+
+template getTransportTooManyError*(code: int = 0): ref TransportTooManyError =
+  let msg =
+    when defined(posix):
+      if code == 0:
+        "Too many open transports"
+      elif code == EMFILE:
+        "[EMFILE] Too many open files in the process"
+      elif code == ENFILE:
+        "[ENFILE] Too many open files in system"
+      elif code == ENOBUFS:
+        "[ENOBUFS] No buffer space available"
+      elif code == ENOMEM:
+        "[ENOMEM] Not enough memory availble"
+      else:
+        "[" & $code & "] Too many open transports"
+    elif defined(windows):
+      case code
+      of 0:
+        "Too many open transports"
+      of ERROR_TOO_MANY_OPEN_FILES:
+        "[ERROR_TOO_MANY_OPEN_FILES] Too many open files"
+      of WSAENOBUFS:
+        "[WSAENOBUFS] No buffer space available"
+      of WSAEMFILE:
+        "[WSAEMFILE] Too many open sockets"
+      else:
+        "[" & $code & "] Too many open transports"
+    else:
+      "[" & $code & "] Too many open transports"
+  newException(TransportTooManyError, msg)
+
+template getConnectionAbortedError*(m: string = ""): ref TransportAbortedError =
+  let msg =
+    if len(m) == 0:
+      "[ECONNABORTED] Connection has been aborted before being accepted"
+    else:
+      "[ECONNABORTED] " & m
+  newException(TransportAbortedError, msg)
+
+template getConnectionAbortedError*(code: int): ref TransportAbortedError =
+  let msg =
+    when defined(posix):
+      if code == 0:
+        "[ECONNABORTED] Connection has been aborted before being accepted"
+      elif code == EPERM:
+        "[EPERM] Firewall rules forbid connection"
+      elif code == ETIMEDOUT:
+        "[ETIMEDOUT] Operation has been timed out"
+      else:
+        "[" & $code & "] Connection has been aborted"
+    elif defined(windows):
+      case code
+      of 0, WSAECONNABORTED:
+        "[ECONNABORTED] Connection has been aborted before being accepted"
+      of WSAENETDOWN:
+        "[ENETDOWN] Network is down"
+      of WSAENETRESET:
+        "[ENETRESET] Network dropped connection on reset"
+      of WSAECONNRESET:
+        "[ECONNRESET] Connection reset by peer"
+      of WSAETIMEDOUT:
+        "[ETIMEDOUT] Connection timed out"
+      else:
+        "[" & $code & "] Connection has been aborted"
+    else:
+      "[" & $code & "] Connection has been aborted"
```

Comment:
The `code` number itself sadly is also platform-specific, so it may have to be combined with a `osErrorMsg` to be useful in isolation (just the number itself is only useful if the underlying platform is known, which is not always provided when someone reports bug)

Reply:
Will create a followup to fix `osErrorMsg` to use `strerror_r`. `strerror` is not thread-safe.

Reply:
Usually i really like to have number, not string which could be confusing because you can't fit everything into one sentence.

Reply:
Followup re `strerror_r`: https://github.com/nim-lang/Nim/pull/19126